tasks/patternpack.js

```diff
@@ -53,18 +53,42 @@ module.exports = function (grunt) {
     }
   };
 
-  function getPackagePathOrFallbackPath(path) {
-    var validatedPath;
-    var pathOfPackage = npmPath + path;
+  function getPathOrPackagePath(path) {
+    var packagePath;
 
-    if (fs.existsSync(pathOfPackage)) {
-      validatedPath = pathOfPackage;
-    } else if (fs.existsSync(path)) {
-      validatedPath = path;
-    } else {
-      throw new Error("Could not be found: " + pathOfPackage + " or " + path);
+    // Attempt to find the configured location
+    if (fs.existsSync(path)) {
+      packagePath = path;
     }
-    return validatedPath;
+
+    // If not found, look for the path as a node package
+    if (!packagePath) {
+      packagePath = getPackagePath(path);
+    }
+
+    if (!packagePath) {
+      throw new Error("Could not find package: " + path);
+    }
+
+    return packagePath;
+  }
+
+  function getPackagePath(name) {
```

Comment:
Enhanced the logic to search for packages.  This has been added to support both the nested npm@2 style of dependencies and the flattened npm@3 style of dependencies.